### PR TITLE
Display KernelCare suggestion based on company id.

### PR DESCRIPTION
**Pull request by estrabd**
Case SWAT-22: Augmented earlier change (59410f93fef) from this
case to look at the company id. If it is cPanel, offer Go link
to purchase KernelCare. If it is not, suggest user contact their
hosting provider.

### CODE REVIEW

File: pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm [hunk 1]
```diff
@@ -50,17 +50,40 @@ sub _suggest_kernelcare {
...
     }
 
     return 1;
 }
 
+# treat cid as a string
+sub _get_companyid {
+    my $companyfile = q{/var/cpanel/companyid};
+    return q{} if not -e $companyfile;
+    open my $fh, "<", $companyfile;
```
Comment (bk2204) [inline pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm]: If this fails for some reason, we'll end up producing a warning on the next line (read on closed filehandle).  What if we did something like this instead?

```perl
my $companyfile = q{/var/cpanel/companyid};
my $cid = q{};
if (open my $fh, "<", $companyfile) {
    $cid = <$fh>;
    close $fh;
}
return $cid;
```

File: pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm [hunk 2]
```diff
@@ -50,17 +50,40 @@ sub _suggest_kernelcare {
...
+        # if cPanel id detected, offer link
+        if ( $companyid eq q{7} ) {
+            $self->add_info_advice(
+                'text' => ['Upgrade to KernelCare'],
+                'suggestion' => [ 'KernelCare provides an easy, effortless way of keeping your operating system kernel up to date without needing to reboot your server. "[output,url,_1,Upgrade to KernelCare,_2,_3]".', 'https://go.cpanel.net/KernelCare', 'target', '_blank', ],
+            );
+        }
+
+        # else, don't offer link
+        else {
```
Comment (bk2204) [inline pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm]: I like that we'll take this branch if there was an error reading the file.  That's a good default.

**Comment by bk2204** [comment]
One suggestion in the pull request about improving error handling of open.

File: pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm (hunk 1 above)
Comment (estrabd) [inline pkg/Cpanel/Security/Advisor/Assessors/Kernel.pm]: This is better, thank you.

**Pull request merged**